review_practice/react-app/src/components/DatePicker/index.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+import 'react-dates/initialize';
+import 'react-dates/lib/css/_datepicker.css';
+import { DayPickerRangeController } from 'react-dates';
+
+import moment from 'moment';
+import 'moment/min/locales';
+
+import * as Style from './style';
+
+export default ({ requestToServer, datePicked, deleteButtonHandle, localeLanguage }) => {
+    moment.locale(localeLanguage);
+
+    const [startDate, setStartDate] = useState(null);
+    const [endDate, setEndDate] = useState(null);
+    const [focusedInput, setFocusedInput] = useState('startDate');
+
+    const onDatesChange = ({ startDate, endDate }) => {
+        datePicked(startDate, endDate);
+
+        setStartDate(startDate);
+        setEndDate(endDate);
+    };
+
+    const onFocusChange = focusedInput => {
+        if (focusedInput === null) setFocusedInput('startDate');
+        else setFocusedInput(focusedInput);
+    };
+
+    const saveButtonHandler = (startDate, endDate) => () => {
+        if (startDate && endDate) requestToServer(startDate, endDate);
+    };
+
+    return (
+        <Style.DatePicker>
+            <DayPickerRangeController
+                startDate={startDate}
+                endDate={endDate}
+                focusedInput={focusedInput}
+                onFocusChange={onFocusChange}
+                onDatesChange={onDatesChange}
+                hideKeyboardShortcutsPanel={true}
+                numberOfMonths={2}
+                isOutsideRange={day => day.isSameOrBefore()}
+                monthFormat="YYYY[년] MM[월]"
+            />
+            <Style.DatePickerButtonWrapper>
+                <a onClick={deleteButtonHandle}>삭제</a>
```

Comment:
```suggestion
                <button type="button" onClick={deleteButtonHandle}>삭제</button>
```

`<A>` 태그는 **하이퍼링크**의 역할을 해야하며, 하이퍼링크가 아닌 단순 버튼 요소는 `<BUTTON>` 요소로 대체할 수 있습니다. 이와 같이 사용하는 경우 웹 접근성(클릭할 수 있는 버튼임을 인지하지 못함)/웹 표준(A요소에는 href가 들어가야함)에 위배됩니다. 버튼 스타일은 간단하게 CSS로 해제가 가능합니다.

참고로 `type="button"` 을 넣어주는 이유는 `<BUTTON>` 의 기본 type은 "submit" 인데, 만약 이 컴포넌트가 폼 엘리먼트 안에 있다면 서버 전송을 실행할 수 있기 때문에, 명시적으로 `type="button"` 을 넣어주는게 좋습니다.